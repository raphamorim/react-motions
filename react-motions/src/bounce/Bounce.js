import React from 'react'
import withBounce from './withBounce'

function Bounce({ children }) {
  return (
    <div>
      {withBounce(children)}
    </div>
  )
}

export default Bounce
