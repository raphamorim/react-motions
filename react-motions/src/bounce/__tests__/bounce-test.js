import React from 'react';
import { mount } from 'enzyme';
import Bounce from '../bounce';

describe('Bounce', () => {
  it('should render', () => {
    const wrapper = shallow(<Bounce>Bounce</Bounce>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<Bounce duration={3}>Bounce</Bounce>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<Bounce fillMode="both">Bounce</Bounce>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should receive transformOrigin prop', () => {
    const wrapper = mount(<Bounce transformOrigin="center bottom">Bounce</Bounce>);
    expect(wrapper.props().transformOrigin).toEqual("center bottom");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<Bounce infinite>Bounce</Bounce>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
