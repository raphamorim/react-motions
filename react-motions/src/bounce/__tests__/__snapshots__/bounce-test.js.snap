// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bounce should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bounce
    duration={1}
    fillMode="both"
    infinite={false}
    transformOrigin="center bottom"
  >
    Bounce
  </Bounce>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Bounce",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "bounce",
        "WebkitTransformOrigin": "center bottom",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "bounce",
        "transformOrigin": "center bottom",
      },
    },
    "ref": null,
    "rendered": "Bounce",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Bounce",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "bounce",
          "WebkitTransformOrigin": "center bottom",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "bounce",
          "transformOrigin": "center bottom",
        },
      },
      "ref": null,
      "rendered": "Bounce",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
